internal/setting/handler.go

```diff
@@ -0,0 +1,312 @@
+			}
+		}
+		handlerutil.WriteJSONResponse(w, http.StatusOK, response)
+		return
+	} else {
+		for i, publicKey := range publicKeys {
+			response[i] = PublicKeyResponse{
+				Title:     publicKey.Title,
+				PublicKey: publicKey.PublicKey,
+			}
+		}
+	}
+
+	handlerutil.WriteJSONResponse(w, http.StatusOK, response)
+}
+
+func (h *Handler) AddUserPublicKeyHandler(w http.ResponseWriter, r *http.Request) {
+	traceCtx, span := h.tracer.Start(r.Context(), "AddUserPublicKeyHandler")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, h.logger)
+
+	user, err := jwt.GetUserFromContext(r.Context())
+	if err != nil {
+		logger.DPanic("Can't find user in context, this should never happen")
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	userId, err := uuid.Parse(user.ID)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	var request AddPublicKeyRequest
+	err = handlerutil.ParseAndValidateRequestBody(traceCtx, h.validator, r, &request)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+	err = validatePublicKey(request.PublicKey)
+	if err != nil {
+		handlerutil.WriteJSONResponse(w, http.StatusBadRequest, err)
+		return
+	}
+
+	publicKey := AddPublicKeyParams{
+		UserID:    userId,
+		Title:     request.Title,
+		PublicKey: request.PublicKey,
+	}
+
+	addedPublicKey, err := h.settingStore.AddPublicKey(traceCtx, publicKey)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	response := PublicKeyResponse{
+		Title:     addedPublicKey.Title,
+		PublicKey: addedPublicKey.PublicKey,
+	}
+	handlerutil.WriteJSONResponse(w, http.StatusOK, response)
+}
+
+func (h *Handler) DeletePublicKeyHandler(w http.ResponseWriter, r *http.Request) {
+	traceCtx, span := h.tracer.Start(r.Context(), "DeletePublicKeyHandler")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, h.logger)
+
+	user, err := jwt.GetUserFromContext(r.Context())
+	if err != nil {
+		logger.DPanic("Can't find user in context, this should never happen")
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	userId, err := uuid.Parse(user.ID)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	var request DeletePublicKeyRequest
+	err = handlerutil.ParseAndValidateRequestBody(traceCtx, h.validator, r, &request)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+	publicKeyId, err := uuid.Parse(request.Id)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	// Check if the public key belongs to the user
+	publicKey, err := h.settingStore.GetPublicKeyById(traceCtx, publicKeyId)
+	if err != nil {
+		problem.WriteError(traceCtx, w, err, logger)
+		return
+	}
+	if publicKey.UserID != userId {
+		logger.Warn("Public key id does not match user id", zap.String("userId", userId.String()), zap.String("publicKeyId", request.Id))
+		handlerutil.WriteJSONResponse(w, http.StatusForbidden, nil)
```

Comment:
Since the user is not the owner of this public key, it should effectively be invisible to them. In this case, it might make more sense and be secure to return a 404 Not Found?